hubblestack/extmods/grains/docker_details.py

```diff
@@ -0,0 +1,76 @@
+"""
+HubbleStack Docker Details Grain.
+CLI Usage - hubble grains.get docker_details
+Example Output - {u'installed': True, u'running': True}
+Author - Mudit Agarwal (muagarwa@adobe.com)
+"""
+import salt.utils.platform
+import logging
+from hubblestack.utils.osquery_lib import query as osquery_util
+log = logging.getLogger(__name__)
+osquery_path = '/opt/osquery/osqueryi'
+
+def get_docker_details(grains):
+  try:
+    docker_grains = {}
+
+    if salt.utils.platform.is_windows():
+      log.debug('This grain is only available on linux')
+      return docker_grains
+
+    docker_details = {}
+    docker_details['installed'] = _is_docker_installed(grains)
+    docker_details['running'] = False
+
+    if docker_details['installed']:
+      docker_details['running'] = _is_docker_process_running()
+
+    log.debug('docker_details = {0}'.format(docker_details))
+
+    docker_grains['docker_details'] = docker_details
+
+    return docker_grains
+  except Exception as e:
+    log.exception('The following exception occurred while fetching docker details {0}'.format(e))
+    return None
+
+
+def _is_docker_installed(grains):
+  try:
+    os_family = grains.get('os_family')
+    if 'coreos' in os_family.lower():
+      return True
```

Comment:
why we are assuming docker to be installed by default in coreos ?

Reply:
Docker cannot be uninstalled on CoreOS since the /usr partition in CoreOS is readonly.

Reply:
salt does not have support for coreos. We will not get os_family as coreos as written in salt grain. 
https://github.com/saltstack/salt/blob/v2019.2.3/salt/grains/core.py#L1431
Please test this behaviour on a coreos machine.

Reply:
osfinger grain reports CoreOS. You can use that. We already use it top.nova like **G@osfinger:*CoreOS*** to execute coreos cis profiles.

Reply:
I had already tested this on CoreOS. Apparently os_family is supported on Coreos and return **u'Container by CoreOS'** as the result.